test/persistence-hooks.suite.js

```diff
@@ -3021,7 +3021,7 @@ module.exports = function(dataSource, should, connectorCapabilities) {
         });
 
         TestModel.upsertWithWhere({ id: existingInstance.id },
-          { id: 'ignored', name: 'new name' },
+          { name: 'new name' },
```

Comment:
Please note for non-atomic implementation `id` change in `data` object does not get ignored; for more info please see: https://github.com/strongloop/loopback-datasource-juggler/blob/72200ce935367cffc7c7df3b6cffc625acb8192a/lib/dao.js#L3020-L3033